prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs

```diff
@@ -1460,16 +1460,19 @@ data SimpleDocStream ann =
 removeTrailingWhitespace :: SimpleDocStream ann -> SimpleDocStream ann
 removeTrailingWhitespace = go (RecordedWhitespace [] 0)
   where
-    commitSpaces
+    commitWhitespace
         :: [Int] -- Withheld lines
         -> Int -- Withheld spaces
         -> SimpleDocStream ann
         -> SimpleDocStream ann
-    commitSpaces [] 0 = id
-    commitSpaces [] 1 = SChar ' '
-    commitSpaces [] n = SText n (T.replicate n " ")
-    commitSpaces [i] n = SLine i . commitSpaces [] n
-    commitSpaces (_:is) n = SLine 0 . commitSpaces is n
+    commitWhitespace is0 n0 = commitLines is0 . commitSpaces n0
+      where
+        commitLines [] = id
+        commitLines (i:is) = foldr (\_ f -> SLine 0 . f) (SLine i) is
```

Comment:
Not sure whether something like this would perform better?!

```suggestion
        commitLines (i:is) sds0 = foldr (\_ sds -> SLine 0 sds) (SLine i sds0) is
```

Reply:
Not sure either. The second version is a bit easier to read though.